tests/std/tests/P0528R3_cmpxchg_pad/test.cpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <atomic>
+#include <cassert>
+#include <cstddef>
+#include <cstring>
+
+struct X1 {
+    char x : 6;
+
+    void set(char v) {
+        x = v;
+    }
+};
+
+struct X2 {
+    short x : 9;
+
+    void set(char v) {
+        x = v;
+    }
+};
+
+#pragma pack(push, 1)
+struct X3 {
+    char x : 4;
+    char : 2;
+    char y : 1;
+    short z;
+
+    void set(char v) {
+        x = v;
+        y = 0;
+        z = 0;
+    }
+};
+#pragma pack(pop)
+
+#pragma warning(push)
+#pragma warning(disable : 4324) // '%s': structure was padded due to alignment specifier
+struct alignas(4) X4 {
+    char x;
+
+    void set(char v) {
+        x = v;
+    }
+};
+#pragma warning(pop)
+
+struct X8 {
+    char x;
+    long y;
+
+    void operator&() const {}

```

Comment:
`= delete;` would make it even clearer that this is intended never to be called.